### DISCUSSION TITLE
I would like an inventory type "directory"

### DISCUSSION THREAD

**Discussion by jdranchman** [semaphoreui/semaphore#1236] [Ideas]
So that I may include whatever inventory file type or dynamic inventory (say vmware) in that directory. This is currently supported by ansible. Common example is to include several dynamic inventory sources into "one".

See: https://docs.ansible.com/ansible/latest/inventory_guide/intro_inventory.html#organizing-inventory-in-a-directory

**Comment by jeremyatourville** [comment]
If I may add to this discussion, I also think it would be useful to be able to create inventory based on resources such as IdM/FreeIPA and RH Satellite/Foreman.  I believe both of these items would be categorized as dynamic inventory sources.  This type of functionality exists in AWX/Ansible Automation Platform.

**Comment by jrglynn2** [comment]
You can do this, it's just labeled "file" ...

So for the Type field, you choose "File", and for the "Path to Inventory File" field, you just enter the path - like `/path/to/dynamic/directory/`

**Reply by dc-jbd** [reply]
Then for my request it is more likely a documentation issue (text in the code too). For the other inventory types maybe separate requests would be better so that can be tracked.

**Comment by craigarms** [comment]
As far as I know using dynamic inventories with Ansible, (I've only used the Azure Dynamic Inventory and Netbox), you always point the ansible-playbook to a YAML file which contains the configuration of the plugin.

So as long as you have the needed collection in your collections/requirements.yml you can point the "File" inventory to the YAML file containing the plugin configuration.

I have this currently setup for Netbox and it works a treat see #1336
